### PR TITLE
#998: Added GET.UNWATCH command support and fix watch related issues

### LINKED ISSUE #998: Add support for `GET.UNWATCH` command
`GET.WATCH` unsubscribes from a `GET.WATCH` subscription.

### PR DESCRIPTION
**Pull request by psrvere**
Closes #998 

## Changes
1. Implemented GET.UNWATCH command in RESP server with integration tests. 
2. Updated Dice SDK version to fix a minor bug

## Bug Fixes
1. Fixed panic in GET.WATCH command when called without arguments
2. Fixed premature fingerprint deletion in `handleUnsubscription` function (`watch_manager.go`). This prevented execution of the next cleanup block.
3. Corrected `cmdWatchSubscriptionChan` initialisation that was incorrectly set to nil in PR #1169

### CODE REVIEW

**Comment by psrvere** [comment]
@JyotinderSingh - Please reivew.

File: internal/worker/worker.go [hunk 1]
```diff
@@ -248,7 +248,37 @@ func (w *BaseWorker) executeCommand(ctx context.Context, diceDBCmd *cmd.DiceDBCm
...
+	}
+
+	// Unsubscribe Unwatch command type
+	if meta.CmdType == Unwatch {
+		command := cmdList[len(cmdList)-1]
+		w.cmdWatchSubscriptionChan <- watchmanager.WatchSubscription{
+			Subscribe:    false,
+			WatchCmd:     command,
+			AdhocReqChan: w.adhocReqChan,
+			Fingerprint:  command.GetFingerprint(),
```
Comment (JyotinderSingh) [inline internal/worker/worker.go:274]: The UNWATCH command unsubscribes using the fingerprint and not the command. The usage is as follows:
```sh
> GET.WATCH k
1) GET
2) 12345 # Fingerprint
3) value
> GET.UNWATCH 12345
OK
```

**Review by JyotinderSingh — CHANGES_REQUESTED** [review]
Thanks for working on this @psrvere! I believe there is a slight misunderstanding on how the command works. I've explained it in my other comment. I don't think it will take too much effort to make that change.

File: internal/worker/worker.go (hunk 1 above)
Comment (psrvere) [reply internal/worker/worker.go]: FIxed.

**Review by JyotinderSingh — APPROVED** [review]
Thanks for the quick resolution to the comments @psrvere! The changes look and well tested. Thanks for making the required SDK changes too.

Approved!

**Pull request merged**